pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -905,3 +905,37 @@ int make_iso_8601_datetime(npy_datetimestruct *dts, char *outstr, int outlen,
                  outlen);
     return -1;
 }
+
+
+int make_iso_8601_timedelta(pandas_timedeltastruct *tds,
```

Comment:
Note that we do also have a isoformat function for time deltas defined here:

https://github.com/pandas-dev/pandas/blob/cdffa43af34aa832105dd3787eec20ea0222af54/pandas/_libs/tslibs/timedeltas.pyx#L1097

But I think the path to get that to work in the JSON module would be way more complicated than implementing in C as we do here (and less consistent with other datetime handling). 

Maybe as a follow up we want to use this C function in the Timedelta class